bot/cogs/antimalware.py

```diff
@@ -38,6 +38,16 @@ class AntiMalware(Cog):
     def __init__(self, bot: Bot):
         self.bot = bot
 
+    def _get_whitelisted_file_formats(self) -> list:
+        """Get the file formats currently on the whitelist."""
+        return [item.get('content') for item in self.bot.allow_deny_list_cache['file_format.True']]
```

Comment:
It's slightly unfortunate you have to use a comprehension every time you need a list of contents. Can the data be restructured to avoid this? In practice, you only need the content and the comments, so caching them in a dict with the former as keys and latter as values may work. Admittedly, it does sound a bit strange, but the contents are guaranteed to be unique already.

Reply:
Hmm, you've got a point. I mean, this is cheap since this list will always be really really short, but we are running this list comprehension for every single message that contains an attachment and isn't sent by a staff member.

I suppose caching may be possible here.

Reply:
By "caching", I meant modifying the structure of the cache you've implemented rather than creating a separate cache.

Reply:
Sorry, I have no idea how you propose we do that.